openc3/python/openc3/accessors/template_accessor.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2024 OpenC3, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+from openc3.accessors.accessor import Accessor
+import re
+
+
+class TemplateAccessor(Accessor):
+    def __init__(self, packet, left_char="<", right_char=">"):
+        super().__init__(packet)
+        self.left_char = left_char
+        self.right_char = right_char
+        self.configured = False
+
+    def configure(self):
+        if self.configured:
+            return
+
+        escaped_left_char = self.left_char
+        if self.left_char == "(":
+            escaped_left_char = f"\\{self.left_char}"
+        escaped_right_char = self.right_char
+        if self.right_char == ")":
+            escaped_right_char = f"\\{self.right_char}"
+
+        # Convert the template into a Regexp for reading each item
+        template = self.packet.template[:]
+        template_items = re.compile(f"{escaped_left_char}.*?{escaped_right_char}", re.X).findall(template.decode())
+        escaped_read_template = re.escape(template.decode())
+
+        self.item_keys = []
+        for item in template_items:
+            self.item_keys.append(item[1:-1])
+            # If they're using parens we have to escape them
+            # since we're working with the already escaped template
+            if self.left_char == "(":
+                item = f"\({item[1:]}"
+            if self.right_char == ")":
+                item = f"{item[0:-1]}\)"
```

Comment:
At line 34 and 37 it is a double `\\` but here a single `\(`. Is that correct? Ruby has double `\\`.

Reply:
Yes.  One is for a regex and the other is for a string replace